### DISCUSSION TITLE
using register_converter() on all ModelViewSet viewsets

### DISCUSSION THREAD

**Discussion by parad0x96** [encode/django-rest-framework#8800] [Question & Answer]
Is there a way to use path converters with ModelViewSet without explicitly writing the path urls ?
For example I have this path converter:

```
from .utils import REGEX
from . import encode_id, decode_id


class HashidsConverter():
    regex = REGEX
    def to_python(self, value: str) -> int:
        return decode_id(value)

    def to_url(self, value: int) -> str:
        return encode_id(value)

```
in `urls.py` file

```
router = DefaultRouter()
router.register("client", ClientModelViewSet)
register_converter(HashidsConverter, "hashid")
urlpatterns = [
    path('admin/', admin.site.urls),
    #path('', include(router.urls)),
    path('client/<hashid:pk>', ClientModelViewSet.as_view({"get":"retrieve"}), name="client")
]
```

**Comment by jerinpetergeorge** [comment]
This is not _yet_ possible in DRF; but there are some existing issues and PRs ( https://github.com/encode/django-rest-framework/issues/6148 and https://github.com/encode/django-rest-framework/pull/6789). Hopefully, we'll get this feature very soon.

**Reply by auvipy** [reply]
wait for that. we will get it soon

**Reply by parad0x96** [reply]
> This is not _yet_ possible in DRF; but there are some existing issues and PRs ( #6148 and #6789). Hopefully, we'll get this feature very soon.

Alright thank you I hope we'll get it soon
